backend/kesaseteli/applications/tests/test_excel_export.py

```diff
@@ -257,6 +282,110 @@ def employer_summer_voucher_sorting_key(voucher: EmployerSummerVoucher):
                 ), excel_field.title
 
 
+@pytest.mark.django_db
+@override_settings(
+    EXCEL_DOWNLOAD_BATCH_SIZE=3,
+    NEXT_PUBLIC_MOCK_FLAG=False,
+)
+def test_youth_excel_download_content(staff_client):  # noqa: C901
+    def youth_application_sorting_key(app: YouthApplication):
+        # Sorting key should be the same as what is used to order by queryset results
+        return app.created_at, app.pk
+
+    InactiveYouthApplicationFactory()  # This should not be exported
+    apps = (
+        ActiveYouthApplicationFactory.create_batch(size=12)
+        + [
+            YouthApplicationFactory(status=status)
+            for status in YouthApplicationStatus.active_values()
+        ]
+        + [
+            ActiveVtjTestCaseYouthApplicationFactory(last_name=vtj_test_case)
+            for vtj_test_case in VtjTestCase.values
+        ]
+    )
+    # Update created_at specially as its initial save uses auto_now_add
+    for app in apps:
+        app.created_at -= timedelta(seconds=random.randint(0, 60 * 60 * 24 * 10))
+        app.save(update_fields=["created_at"])
+        app.refresh_from_db()
+    assert min(app.created_at for app in apps) != max(app.created_at for app in apps)
+    apps = sorted(apps, key=youth_application_sorting_key)
+
+    with translation.override("en"):  # Should still use Finnish translations
+        response = staff_client.get(youth_excel_download_url())
+
+    assert isinstance(response, StreamingHttpResponse)
+
+    workbook = openpyxl.load_workbook(filename=BytesIO(response.getvalue()))
+    active_worksheet = workbook.active
+
+    rows_generator = active_worksheet.rows
+    header_row = next(rows_generator)
+    data_rows = [next(rows_generator) for _ in apps]
+    with pytest.raises(StopIteration):
+        next(rows_generator)
+
+    output_column_names = tuple(column.value for column in header_row)
+    with translation.override("fi"):  # Make extra sure to test for Finnish column names
+        expected_output_columns_names = (
+            YouthApplicationExcelExportView.output_column_names()
+        )
+    assert output_column_names == expected_output_columns_names
+
+    source_fields = YouthApplicationExcelExportView.source_fields()
+    for data_row, app in zip(data_rows, apps):
+        for output_cell, source_field in zip(data_row, source_fields):
+            output_value = output_cell.value
+            if source_field == "application_date":
```

Comment:
Does python support switch-statement instead of multiple ifs? I think it would be more readable

Reply:
There's no switch statement, but if we used Python 3.10 or later we could perhaps use [match/case](https://docs.python.org/3/whatsnew/3.10.html#pep-634-structural-pattern-matching) but [we're using Python 3.8](https://github.com/City-of-Helsinki/yjdh/blob/5f932b6c25499f9c8ef8aa7827e72b612a6c9573/backend/docker/kesaseteli.Dockerfile#L2). Also some kind of a dictionary lookup could perhaps be used instead and the checks be put into functions but it depends on the code whether that would be readable or not. Having a lot of if/elif/else statements and nested ones in them does make the logic a bit hard to follow.